### ISSUE TITLE
Add IgnoredCops configurable attribute to DisableCopsWithinSourceCodeDirective

### ISSUE THREAD

**Issue by sethpuckett** [rubocop-hq/rubocop#9399]
## Is your feature request related to a problem? Please describe.

`DisableCopsWithinSourceCodeDirective` is great for ensuring problems don't get swept under the rug. But only having the option to enable it for all directives or none is limiting. There are situations when it is reasonable to disable certain types of directives on a per-line or per-method basis.

## Describe the solution you'd like

An `IgnoredCops` configurable attribute to allow some directives to be enabled/disabled with comments.

```
Style/DisableCopsWithinSourceCodeDirective:
  Enabled: true
  IgnoredCops: ['Metrics/MethodLength', 'Metrics/AbcSize']
```

**Examples:**

`Ignored Cops: [] (default)`

```ruby
# bad

# rubocop:disable Metrics/MethodLength, Metrics/AbcSize
def map_big_hash
  {
    attr_1: 'test',
    attr_2: 'test',
    ...
    attr_30: 'test'
  }
end
# rubocop:enable Metrics/MethodLength, Metrics/AbcSize
```

`Ignored Cops: ['Metrics/MethodLength', 'Metrics/AbcSize']`

```ruby
# good

# rubocop:disable Metrics/MethodLength, Metrics/AbcSize
def map_big_hash
  {
    attr_1: 'test',
    attr_2: 'test',
    ...
    attr_30: 'test'
  }
end
# rubocop:enable Metrics/MethodLength, Metrics/AbcSize
```

## Describe alternatives you've considered

The alternative is to disable the directive or to rely on whole-file exclusions, which is not ideal.

**Comment by dvandersluis** [comment]
`AllowedCops` would be a better name to avoid the double negative.
